Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -0,0 +1,182 @@
+                _RingtoneNoteArgumentArray[noteCount] = (Note)Enum.Parse(typeof(Note), _currentNoteString);
+                }
+                catch
+                {
+                    return;
+                }
+            }
+
+            UpdateRingerComponent(ringer, _RingtoneNoteArgumentArray);
+        }
+
+        public void RandomizeRingtone(EntityUid uid, RingerComponent ringer, ComponentInit args) //Randomizes the ringtone
+        {
+            Note[] _randomizedRingerNoteArr = new Note[4];
+            var randomNote = new Random();
+
+            for(int count = 0; count < 4; count++)
+            {
+                int randomNoteIndex = randomNote.Next(Enum.GetNames(typeof(Note)).Length - 1);
+                var currentNote = (Note)randomNoteIndex;
+                _randomizedRingerNoteArr[count] = currentNote;
+            }
+
+            UpdateRingerComponent(ringer, _randomizedRingerNoteArr);
+
+        }
+
+        //Non Event Functions
+
+        private bool UpdateRingerComponent(RingerComponent ringer, Note[] ringerNoteArr)
+        {
+            string[] _ringtonePathArr = new string[4];
+
+            for(int strngCount = 0; strngCount < 4; strngCount++)
+            {
+
+                if((int)ringerNoteArr[strngCount] < 12 )
+                {
+                    _ringtonePathArr[strngCount] = _pathIndex[ringerNoteArr[strngCount].ToString()];
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            ringer.RingtoneStringArr = ConvertToString(ringerNoteArr);
+            ringer.SetRingerAudioPaths = _ringtonePathArr;
+            UpdateRingerUserInterface(ringer);
+
+            return true;
+        }
+
+        private void UpdateRingerUserInterface(RingerComponent ringer)
+        {
+            var ui = ringer.Owner.GetUIOrNull(RingerUiKey.Key);
+            ui?.SetState(new RingerUpdateState(ringer.IsPlaying, ringer.RingtoneStringArr, ringer.SetRingerAudioPaths));
+        }
+
+        public bool OpenRingerUI(RingerComponent ringer, IPlayerSession session)
+        {
+            var ui = ringer.Owner.GetUIOrNull(RingerUiKey.Key);
+            ui?.Toggle(session);
+            return true;
+        }
+
+        public override void Update(float frameTime) //Responsible for actually playing the ringtone
+        {
+            foreach(var ringer in EntityManager.EntityQuery<RingerComponent>())
+            {
+                if(ringer.IsPlaying)
+                {
+                    ringer.TimeElapsed += frameTime;
+
+                    if(ringer.TimeElapsed > 0.200)
+                    {
+                        if(ringer.NoteCount > 3)
+                        {
+                            ringer.IsPlaying = false;
+                            UpdateRingerUserInterface(ringer);
+                            ringer.TimeElapsed = 0;
```

Comment:
```suggestion
                            ringer.TimeElapsed -= 0.2;
```

You're going to get timing issues otherwise.